app/domain/authentication/authn_k8s/authentication_request.rb

```diff
@@ -0,0 +1,32 @@
+module Authentication
+  module AuthnK8s
+
+    # This class validates the k8s authentication request against the resource
+    # restrictions configured in the policy for the requesting host.
+    # It uses the 'validate_k8s_resource' command-class to validate a k8s
+    # resource against the k8s API.
+    # This command-class receives the resource's type and name as input.
+    class AuthenticationRequest
+      def initialize(namespace:, validate_k8s_resource:)
+        @namespace = namespace
+        @validate_k8s_resource = validate_k8s_resource
+      end
+
+      def valid?(restriction_name, restriction_value)
```

Comment:
the name `valid?` is a bit weird for me. I would expect the method `valid?` in the `AuthenticationRequest` class to tell me if the authentication request is valid but this is not what is happening here. This method says if the restriction is valid in the context of the authentication request. So i think that
`def valid_restriction?(restriction_name, restriction_value)` is better.
you can also go with
`def valid_restriction?(name, value)`

Reply:
I like it!

Reply:
done